compaction/compaction_manager.cc

```diff
@@ -1009,11 +1009,11 @@ future<> compaction_manager::remove(column_family* cf) {
     // The requirement above is provided by stop_ongoing_compactions().
     _postponed.erase(cf);
 
-    // Wait for all functions running under gate to terminate.
-    auto close_gate = _compaction_state[cf].gate.close();
-
     // Wait for the termination of an ongoing compaction on cf, if any.
-    co_await when_all_succeed(stop_ongoing_compactions("column family removal", cf), std::move(close_gate));
+    co_await stop_ongoing_compactions("column family removal", cf);
```

Comment:
`stop_ongoing_compactions` is called during normal operation (e.g. via the api), not only during shutdown or `table::stop`.
Therefore we don't always want to close the gate.
It is the responsibility of the caller to prevent further compactions from happening, if required.

For example:
- In the `compaction_manager::remove` case, detaching the table's `compaction_state` from the compaction_manager _compaction_state map prevents new compactions from starting. And then the gate is closed.

- compaction_manager `stop_compaction` api stops compaction by type but it doesn't prevent other compactions from starting (it is not using stop_ongoing_compactions at the moment, but my plan after this PR is to make it use it)

- `table::disable_auto_compaction` is also planned to use `stop_ongoing_compactions` (See https://github.com/scylladb/scylla/issues/9313) and it uses a state flag on the table to prevent regular compactions from starting during and after stop_ongoing_compactions.